src/java/htsjdk/samtools/util/SequenceUtil.java

```diff
@@ -103,21 +103,40 @@ public static double calculateGc(final byte[] bases) {
     }
 
     /**
-     * Throws an exception only if both parameters are not null
-     *
+     * default signature that forces the lists to be the same size
      * @param s1 a list of sequence headers
      * @param s2 a second list of sequence headers
      */
     public static void assertSequenceListsEqual(final List<SAMSequenceRecord> s1, final List<SAMSequenceRecord> s2) {
+        assertSequenceListsEqual(s1, s2, false);
+    }
+        /**
+         * Throws an exception only if both parameters are not null
+         * allows to only check that one list is a prefix of the other.
+         *
+         * @param s1 a list of sequence headers
+         * @param s2 a second list of sequence headers
+         * @param checkPrefixOnly a flag specifying whether to only look at the first records in the lists
+         */
+    public static void assertSequenceListsEqual(final List<SAMSequenceRecord> s1, final List<SAMSequenceRecord> s2, boolean checkPrefixOnly) {
         if (s1 != null && s2 != null) {
 
-            if (s1.size() != s2.size()) {
-                throw new SequenceListsDifferException(
-                        "Sequence dictionaries are not the same size (" + s1.size() + ", " + s2.size() +
-                                ")");
-            }
+            final int sizeToTest;
 
-            for (int i = 0; i < s1.size(); ++i) {
+            if (checkPrefixOnly) {
+                sizeToTest = Math.min(s1.size(), s2.size());
+                if (sizeToTest == 0) {
+                    throw new SequenceListsDifferException("At least one of the dictionaries must be not empty.");
```

Comment:
Sorry, I missed one before.  The error message here should be:
"Neither of the dictionaries can be empty."